src/common/map_cacher.hpp

```diff
@@ -27,7 +27,7 @@ class Transaction {
 public:
   /// Set keys according to map
   virtual void set_keys(
-    const std::map<K, V> &keys ///< [in] keys/values to set
+    const std::map<K, V, std::less<>> &keys ///< [in] keys/values to set
```

Comment:
The transparent comparator definitely will help cases when `K` is `std::string`. However, IIUC, if `std::string` hadn't provided the operator overload for `char*`, then the `std::less<>` opt-in would have resulted in doing the conversion on every comparison instead of just once – on a call to `std::map<K, V>::find()`.

If so, maybe we could put `static_assert` ensuring `K` is one of *allowed (enlisted) types*? On the other hand, such *index of Ks* would introduce tremendous amount of coupling.

Reply:
yeah. makes sense! will do.